hbase-it/src/test/java/org/apache/hadoop/hbase/chaos/factories/MobNoKillMonkeyFactory.java

```diff
@@ -75,7 +75,7 @@
 
     Action[] actions4 = new Action[] { new DumpClusterStatusAction() };
 
-    return new PolicyBasedChaosMonkey(util,
+    return new PolicyBasedChaosMonkey(properties, util,
```

Comment:
Do we really need to specify extra properties when creating policy for all these factories? Otherwise, if this is just duo to the change in PolicyBasedChaosMonkey constructor, it would benefit from overloading the constructor there.

Reply:
I kind of had to. The issue was that Actions do not have access to these properties. This is the main reason we have this clutter in factories where we read the properties and pass them along to the actions in the action constructor, when it could have been an internal thing. So I looked for a way to get these properties to them. (It was a minor issue that Action.init could not use these properties and had to rely on the default values from the config. Passing it along fixed this too.) Anyway the best way to do this was through PolicyBasedChaosMonkey and PolicyContext. Technically I could have just created a setter on PolicyBasedChaosMonkey and leave the constructor alone and call the setter from IntegrationTestBase, but it felt like a bad practice to circumvent a factory.